spec/lib/tasks_helpers/metrics_spec.rb

```diff
@@ -0,0 +1,102 @@
+describe 'Metrics' do
+  let(:header) do
+    %w[
+      tenant
+      portfolio_count
+      product_count
+      portfolio_share_count
+    ].collect(&:titleize)
+  end
+
+  let(:tenant) do
+    FactoryBot.create(
+      :tenant,
+      :name            => 'tenant_portfolio_shared_context',
+      :external_tenant => 'external_tenant_portfolio_shared'
+    )
+  end
+
+  def create_portfolio
+    portfolio = FactoryBot.create(:portfolio)
+    FactoryBot.create(:access_control_entry, :aceable => portfolio)
+    portfolio.update_metadata
+  end
+
+  def create_shared_portfolio
+    portfolio = FactoryBot.create(:portfolio)
+    FactoryBot.create(
+      :access_control_entry,
+      :has_read_and_update_permission,
+      :aceable => portfolio
+    )
+    portfolio.update_metadata
+  end
+
+  def create_portfolio_for_tenant
+    ActsAsTenant.with_tenant(tenant) { create_portfolio }
+  end
+
+  def create_shared_portfolio_for_tenant
+    ActsAsTenant.with_tenant(tenant) { create_shared_portfolio }
+  end
+
+  describe '`.portfolio_count`' do
+    it 'returns metric' do
+      FactoryBot.create(:portfolio)
```

Comment:
This should be pulled into a `before` setup block, as well as changed to `create(:portfolio)`